File: clocks/analog_clock.py
from interfaces import BaseAnalogClock
from consts import DayNightDivision


class AnalogClock(BaseAnalogClock):
    def __init__(self):
        self.year = 0
        self.month = 0
        self.day = 0
        self.hour_angle = 0.0
        self.minute_angle = 0.0
        self.second_angle = 0.0
        self.day_night_division = DayNightDivision.AM

    def set_date_time(self, year: int, month: int, day: int, hour_angle: float, minute_angle: float,
                      second_angle: float, day_night_division: DayNightDivision):
        self.year = year
        self.month = month
        self.day = day
        self.hour_angle = hour_angle
        self.minute_angle = minute_angle
        self.second_angle = second_angle
        self.day_night_division = day_night_division

    def get_hour_angle(self) -> float:
        return self.hour_angle

    def get_minute_angle(self) -> float:
        return self.minute_angle

    def get_second_angle(self) -> float:
        return self.second_angle

    def get_year(self) -> int:
        return self.year

    def get_month(self) -> int:
        return self.month

    def get_day(self) -> int:
        return self.day

File: AdapterPattern/AnalogClock.cs
namespace AdapterPattern;

public class AnalogClock : IBaseAnalogClock
{
    private int year;
    private int month;
    private int day;
    private float hourAngle;
    private float minuteAngle;
    private float secondAngle;
    private DayNightDivision dayNightDivision;

    public AnalogClock()
    {
        year = 0;
        month = 0;
        day = 0;
        hourAngle = 0.0f;
        minuteAngle = 0.0f;
        secondAngle = 0.0f;
        dayNightDivision = DayNightDivision.AM;
    }

    public void SetDateTime(int year, int month, int day, float hourAngle, float minuteAngle, float secondAngle, DayNightDivision dayNightDivision)
    {
        this.year = year;
        this.month = month;
        this.day = day;
        this.hourAngle = hourAngle;
        this.minuteAngle = minuteAngle;
        this.secondAngle = secondAngle;
        this.dayNightDivision = dayNightDivision;
    }

    public float GetHourAngle()
    {
        return hourAngle;
    }

    public float GetMinuteAngle()
    {
        return minuteAngle;
    }

    public float GetSecondAngle()
    {
        return secondAngle;
    }

    public int GetYear()
    {
        return year;
    }

    public int GetMonth()
    {
        return month;
    }

    public int GetDay()
    {
        return day;
    }

    public DayNightDivision GetDayNightDivision()
    {
        return dayNightDivision;
    }
}
